### PR TITLE
Task Library: Add more tasks to the pip package

### PR DESCRIPTION
**Pull request by kinaryml**
To reproduce locally, build the Pypi wheel by
```
bazel build -c opt tensorflow_lite_support/tools/pip_package:build_pip_package
./bazel-bin/tensorflow_lite_support/tools/pip_package/build_pip_package --dst wheels --nightly_flag
```

Code snippet for vision tasks:
```python
from tflite_support.task import vision
from tflite_support.task import core
from tflite_support.task import processor

# ImageClassifier
base_options = core.BaseOptions(file_name=model_path)
classification_options = processor.ClassificationOptions(max_results=2)
options = vision.ImageClassifierOptions(base_options=base_options, classification_options=classification_options)
classifier = vision.ImageClassifier.create_from_options(options)
# classifier = vision.ImageClassifier.create_from_file(model_path)
image = vision.TensorImage.create_from_file(image_path)
image_result = classifier.classify(image)
print(image_result)

# ObjectDetector
base_options = core.BaseOptions(file_name=model_path)
detection_options = processor.DetectionOptions(max_results=2)
options = vision.ObjectDetectorOptions(base_options=base_options, detection_options=detection_options)
detector = vision.ObjectDetector.create_from_options(options)
# detector = vision.ObjectDetector.create_from_file(model_path)
image = vision.TensorImage.create_from_file(image_path)
image_result = detector.detect(image)
print(image_result)

# ImageEmbedder
base_options = core.BaseOptions(file_name=model_path)
embedding_options = processor.EmbeddingOptions(l2_normalize=True)
options = vision.ImageEmbedderOptions(base_options=base_options, embedding_options=embedding_options)
embedder = vision.ImageEmbedder.create_from_options(options)
# embedder = vision.ImageEmbedder.create_from_file(model_path)
image = vision.TensorImage.create_from_file(image_path)
image_crop = vision.TensorImage.create_from_file(image_path_crop)
image_result = embedder.embed(image)
crop_result = embedder.embed(image_crop)
image_feature_vector = image_result.embeddings[0].feature_vector
crop_feature_vector = crop_result.embeddings[0].feature_vector
similarity = embedder.cosine_similarity(image_feature_vector, crop_feature_vector)
print(similarity)

# ImageSegmenter
base_options = core.BaseOptions(file_name=model_path)
segmentation_options = processor.SegmentationOptions(
    output_type=processor.SegmentationOptions.OutputType.CATEGORY_MASK)
options = vision.ImageSegmenterOptions(base_options=base_options, segmentation_options=segmentation_options)
segmenter = vision.ImageSegmenter.create_from_options(options)
# segmenter = vision.ImageSegmenter.create_from_file(model_path)
image_file = vision.TensorImage.create_from_file(image_path)
segmentation_result = segmenter.segment(image_file)
print(segmentation_result)
```

Code snippet for audio tasks
```python
from tflite_support.task import audio
from tflite_support.task import core
from tflite_support.task import processor

# AudioClassifier
base_options = core.BaseOptions(file_name=model_path)
classification_options = processor.ClassificationOptions(max_results=2)
options = audio.AudioClassifierOptions(base_options=base_options, classification_options=classification_options)
classifier = audio.AudioClassifier.create_from_options(options)
# classifier = audio.AudioClassifier.create_from_file(model_path)
audio_file = audio.TensorAudio.create_from_wav_file(audio_path, classifier.required_input_buffer_size)
audio_result = classifier.classify(audio_file)
print(audio_result)

# AudioEmbedder
base_options = core.BaseOptions(file_name=model_path)
embedding_options = processor.EmbeddingOptions(l2_normalize=True)
options = audio.AudioEmbedderOptions(base_options=base_options, embedding_options=embedding_options)
embedder = audio.AudioEmbedder.create_from_options(options)
# embedder = audio.AudioEmbedder.create_from_file(model_path)
audio_file0 = audio.TensorAudio.create_from_wav_file(audio_path0, classifier.required_input_buffer_size)
audio_file1 = audio.TensorAudio.create_from_wav_file(audio_path1, classifier.required_input_buffer_size)
audio_result0 = embedder.embed(audio_file0)
audio_result1 = embedder.embed(audio_file1)
audio_feature_vector0 = audio_result0.embeddings[0].feature_vector
audio_feature_vector1 = audio_result1.embeddings[0].feature_vector
similarity = embedder.cosine_similarity(image_feature_vector, crop_feature_vector)
print(similarity)
```

Code snippet for text tasks
```python
from tflite_support.task import text
from tflite_support.task import core
from tflite_support.task import processor

# TextEmbedder
base_options = core.BaseOptions(file_name=model_path)
embedding_options = processor.EmbeddingOptions(l2_normalize=True)
options = text.TextEmbedderOptions(base_options=base_options, embedding_options=embedding_options)
embedder = text.TextEmbedder.create_from_options(options)
# embedder = text.TextEmbedder.create_from_file(model_path)
result0 = embedder.embed(text0)
result1 = embedder.embed(text1)
feature_vector0 = result0.embeddings[0].feature_vector
feature_vector1 = result1.embeddings[0].feature_vector
similarity = embedder.cosine_similarity(feature_vector0, feature_vector1)
print(similarity)
```

### CODE REVIEW

**Comment by khanhlvg** [comment]
@kinaryml You also need to add sounddevice to the list of dependencies [here](https://github.com/tensorflow/tflite-support/blob/7188e5bf3b8679d3eb1ffeaaea852a519260cfb3/tensorflow_lite_support/tools/pip_package/setup.py#L57).

What was the issue you had when installing 'protobuf >= 3.18.0'?

**Comment by kinaryml** [comment]
> @kinaryml You also need to add sounddevice to the list of dependencies [here](https://github.com/tensorflow/tflite-support/blob/7188e5bf3b8679d3eb1ffeaaea852a519260cfb3/tensorflow_lite_support/tools/pip_package/setup.py#L57).
> 
> What was the issue you had when installing 'protobuf >= 3.18.0'?

It ran into a segmentation fault with the newest protobuf pypi package and reverting to an older version (3.18) helped alleviate the bug. Regarding sounddevice, I'm not sure which minor and major version to use. I thought it'd be best if you can let me know what version to use for it.

**Comment by kinaryml** [comment]
Code to reproduce the issue can be found in the Colab linked in this [issue](https://github.com/tensorflow/tflite-support/issues/790#issuecomment-1099015066).

**Comment by khanhlvg** [comment]
For sounddevice, please use `>=0.4.4` (latest version). For the protobuf issue when using newer version, I'll ask Lu for help so please leave the version as-is.

File: tensorflow_lite_support/tools/pip_package/setup.py [hunk 1]
```diff
@@ -55,6 +55,7 @@
     # issue is resolved.
     'flatbuffers >= 1.12, <2',
     'protobuf >= 3.18.0',
+    'sounddevice >= 0.4.4',
```
Comment (kinaryml) [inline tensorflow_lite_support/tools/pip_package/setup.py:58]: @khanhlvg I've added the dependency now.

**Comment by khanhlvg** [comment]
@kinaryml TFLite Support indeed using protobuf 3.18.0 to build the libraries so the setup.py file does need to fix the version of protobuf rather than using `>=`. Could you update accordingly?
https://github.com/tensorflow/tflite-support/blob/acacbc661b55613e17db56f811db414732e078d0/WORKSPACE#L153

File: tensorflow_lite_support/tools/pip_package/setup.py [hunk 2]
```diff
@@ -54,7 +54,8 @@
     # TODO(b/187981032): remove the constraint for 2.0 once the incompatibile
     # issue is resolved.
     'flatbuffers >= 1.12, <2',
-    'protobuf >= 3.18.0',
+    'protobuf == 3.18.0',
```
Comment (kinaryml) [inline tensorflow_lite_support/tools/pip_package/setup.py:57]: @khanhlvg I've updated it to use protobuf==3.18 now.

**Comment by lu-wang-g** [comment]
Thanks for adding all these tasks into the Pypi package, Kinar! Could you please add sample code to use each task just like what I did in the PR description of https://github.com/tensorflow/tflite-support/commit/b1f13f0cb035140de9eebecfb46e7b39e3b1c9c8? Many thanks!

**Comment by kinaryml** [comment]
> Thanks for adding all these tasks into the Pypi package, Kinar! Could you please add sample code to use each task just like what I did in the PR description of [b1f13f0](https://github.com/tensorflow/tflite-support/commit/b1f13f0cb035140de9eebecfb46e7b39e3b1c9c8)? Many thanks!

I've updated the description to show usages for different tasks.

**Review by khanhlvg — APPROVED** [review]

**Pull request merged**